File: android/app/src/main/java/com/guitaripod/pixie/data/model/EditingModels.kt
package com.guitaripod.pixie.data.model

import android.net.Uri

data class SelectedImage(
    val uri: Uri,
    val displayName: String? = null
)

sealed class ToolbarMode {
    object Generate : ToolbarMode()
    data class Edit(val selectedImage: SelectedImage) : ToolbarMode()
}

sealed class EditMode {
    object Simple : EditMode()
}

data class EditOptions(
    val prompt: String = "",
    val variations: Int = 1,
    val size: ImageSize = ImageSize.AUTO,
    val quality: ImageQuality = ImageQuality.LOW,
    val fidelity: FidelityLevel = FidelityLevel.LOW
)

enum class FidelityLevel(val value: String, val displayName: String, val description: String) {
    LOW("low", "Low", "More creative freedom"),
    HIGH("high", "High", "Preserve details (faces, logos)")
}

data class EditToolbarState(
    val isExpanded: Boolean = false,
    val showAdvancedOptions: Boolean = false
)

File: iOS/Pixie/Pixie/Data/Models/EditingModels.swift
import Foundation
import UIKit

struct SelectedImage {
    let image: UIImage
    let url: URL?
    let displayName: String?
}

enum ToolbarMode {
    case generate
    case edit(selectedImage: SelectedImage)
}

struct EditOptions {
    var prompt: String = ""
    var variations: Int = 1
    var size: ImageSize = .auto
    var quality: ImageQuality = .low
    var fidelity: FidelityLevel = .low
    var background: String? = "auto"
    var outputFormat: String = "png"
    var compression: Int? = nil
}

enum FidelityLevel: CaseIterable {
    case low, high
    
    var value: String {
        switch self {
        case .low: return "low"
        case .high: return "high"
        }
    }
    
    var displayName: String {
        switch self {
        case .low: return "Low"
        case .high: return "High"
        }
    }
    
    var description: String {
        switch self {
        case .low: return "More creative freedom"
        case .high: return "Preserve details (faces, logos)"
        }
    }
}

struct EditToolbarState {
    var isExpanded: Bool = false
    var showAdvancedOptions: Bool = false
}